### PR TITLE
ODIS Quota take cEUR balance into account

### PR DESCRIPTION
**Pull request by emilhotkowski**
### Description

We want to also recognize cEUR as a valid balance during quote calculation.

### Tested

I added separate test for every stable token.

### Related issues

- Implements #7532

### Backwards compatibility

It is compatible since we only add cEUR. If person had enough quote before this change they still would. This is something extra. Doesn't event require all signers at the same time to have this change.

### CODE REVIEW

File: packages/phone-number-privacy/combiner/package.json [hunk 1]
```diff
@@ -24,9 +24,9 @@
     "db:migrate": "NODE_ENV=dev ts-node scripts/run-migrations.ts"
   },
   "dependencies": {
-    "@celo/contractkit": "1.0.2",
+    "@celo/contractkit": "1.2.2-dev",
```
Comment (codyborn) [inline packages/phone-number-privacy/combiner/package.json]: Unfortunately the combiner cannot rely upon an unpublished version. Google functions won't allow it.

Comment (emilhotkowski) [reply packages/phone-number-privacy/combiner/package.json:27]: Should I change version in all places to something published but still with cEUR in contractkit ?

File: packages/phone-number-privacy/signer/src/signing/query-quota.ts [hunk 2]
```diff
@@ -272,6 +282,33 @@ export async function getDollarBalance(logger: Logger, ...addresses: string[]):
   })
 }
 
+export async function getEuroBalance(logger: Logger, ...addresses: string[]): Promise<BigNumber> {
```
Comment (codyborn) [inline packages/phone-number-privacy/signer/src/signing/query-quota.ts]: Given that this is similar to `getDollarBalance` we can create a new method `getStableTokenBalance` that receives the token type as a parameter.

**Review by codyborn — APPROVED** [review]
Looks good! Just left a small comment.

File: packages/phone-number-privacy/signer/src/signing/query-quota.ts (hunk 2 above)
Comment (emilhotkowski) [reply packages/phone-number-privacy/signer/src/signing/query-quota.ts]: Done

File: packages/phone-number-privacy/signer/src/signing/query-quota.ts [hunk 3]
```diff
@@ -266,7 +281,7 @@ export async function getDollarBalance(logger: Logger, ...addresses: string[]):
   ).then((values) => {
     logger.trace(
       { addresses, balances: values.map((bn) => bn.toString()) },
-      'Fetched cusd balances for addresses'
+      'Fetched ceur balances for addresses'
```
Comment (alecps) [inline packages/phone-number-privacy/signer/src/signing/query-quota.ts]: Can you parameterize this with the name of the stable token? Something like `Fetched ${stableToken.name} balances for addresses`

**Review by alecps — APPROVED** [review]
Looks good! 💪

File: packages/phone-number-privacy/common/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@celo/phone-number-privacy-common",
-  "version": "1.0.33",
+  "version": "1.0.34",
```
Comment (alecps) [inline packages/phone-number-privacy/common/package.json]:
```suggestion
  "version": "1.0.34-dev",
```

**Review by alecps — CHANGES_REQUESTED** [review]

**Review by alecps — APPROVED** [review]

**Pull request merged**